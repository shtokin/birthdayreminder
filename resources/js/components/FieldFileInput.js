import React, {Component} from 'react'
import {Image} from "semantic-ui-react";

export default class FieldFileInput  extends Component{
  constructor(props) {
    super(props);
    this.onChange = this.onChange.bind(this);
    this.fileInputRef = React.createRef();
    this.state = {
      file: null
    }
  }

  onChange(e) {
    const { input: { onChange } } = this.props;
    onChange(e.target.files[0]);
    this.setState({
      file: URL.createObjectURL(e.target.files[0])
    })
  }

  render(){
    const { input: { value } } = this.props;
    const {input,label, required, meta, photoUrl } = this.props;

    let photoSrc = this.state.file ? this.state.file : photoUrl;
    return(
      <div>
        <label className="ui icon button" onClick={() => this.fileInputRef.current.click()}>
          <i className="file icon"></i>{label}
        </label>
        <Image src={photoSrc} size='large' className="rounded" />
        <input
          ref={this.fileInputRef}
          style={{display: 'none'}}
          type='file'
          accept='.jpg, .png, .jpeg'
          onChange={this.onChange}
        />
      </div>
    )
  }
}