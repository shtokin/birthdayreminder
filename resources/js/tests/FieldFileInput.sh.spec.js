import renderer from "react-test-renderer";
import FieldFileInput from "../components/FieldFileInput";
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from "react";

//configure({ adapter: new Adapter() });

describe('FieldFileInput', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<FieldFileInput />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // const checkbox = shallow(<FieldFileInput label="LAB" />);
    //
    // expect(checkbox.text()).toEqual('LAB');

    //checkbox.find('input').simulate('change');

    //expect(checkbox.text()).toEqual('On');

  });
});