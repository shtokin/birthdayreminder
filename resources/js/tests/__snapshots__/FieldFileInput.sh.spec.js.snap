// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldFileInput snapshot renders 1`] = `
<div>
  <label
    className="ui icon button"
    onClick={[Function]}
  >
    <i
      className="file icon"
    />
  </label>
  <img
    className="ui large image rounded"
  />
  <input
    accept=".jpg, .png, .jpeg"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</div>
`;
